### PR TITLE
Callout: use generated props table in docs

### PR DESCRIPTION
**Pull request by dangerismycat**
see title

### CODE REVIEW

File: docs/pages/callout.js [hunk 1]
```diff
@@ -1,431 +1,375 @@
...
+import docgen, { type DocGen } from '../components/docgen.js';
 
-const cards: Array<Node> = [];
-const card = (c) => cards.push(c);
-
-card(
-  <PageHeader
-    name="Callout"
-    description={`
+export default function SearchFieldPage({ generatedDocGen }: {| generatedDocGen: DocGen |}): Node {
```
Comment (AlbertCarreras) [inline docs/pages/callout.js:9]: CalloutPage

File: packages/gestalt/src/Callout.js [hunk 2]
```diff
@@ -12,15 +12,72 @@ import typography from './Typography.css';
...
+  /**
+   * Main content of Callout. Content should be [localized](https://gestalt.pinterest.systems/callout#Localization).
+   *
+   * See [Best Practices](https://gestalt.pinterest.systems/callout#Best-practices) for more info.
+   */
   message: string,
-  primaryAction?: ActionDataType,
-  secondaryAction?: ActionDataType,
+  /**
+   * Main action for users to take on Upsell. If `href` is supplied, the action will serve as a link. See [OnLinkNavigationProvider](https://gestalt.pinterest.systems/callout/OnLinkNavigationProvider) to learn more about link navigation.
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Callout.js:35]: Main action for users to take on Callout

File: packages/gestalt/src/Callout.js [hunk 3]
```diff
@@ -12,15 +12,72 @@ import typography from './Typography.css';
...
+      | SyntheticMouseEvent<HTMLAnchorElement>
+      | SyntheticKeyboardEvent<HTMLAnchorElement>
+      | SyntheticKeyboardEvent<HTMLButtonElement>,
+      {| disableOnNavigation: () => void |},
+    >,
+    rel?: 'none' | 'nofollow',
+    target?: null | 'self' | 'blank',
+  |},
+  /**
+   * Secondary action for users to take on Upsell. If `href` is supplied, the action will serve as a link. See [OnLinkNavigationProvider](https://gestalt.pinterest.systems/callout/OnLinkNavigationProvider) to learn more about link navigation.
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Callout.js:55]: Secondary action for users to take on Callout

File: docs/pages/callout.js (hunk 1 above)
Comment (dangerismycat) [reply docs/pages/callout.js:9]: good catch!

File: packages/gestalt/src/Callout.js (hunk 2 above)
Comment (dangerismycat) [reply packages/gestalt/src/Callout.js:35]: this is currently in the docs 😱 🤦‍♂️

**Pull request merged**